File: 02-frontend/src/app/model/invoice.ts
import {PolicyPart} from "./policypart";

export class Invoice {
  public id: number;
  public invoiceNumber: string;
  public text: string;
  public conclusionDate: Date;
  public maxDate: Date;
  public paidDate: Date;

  constructor() {
    this.id = 0;
    this.invoiceNumber = '';
    this.text = '';
    this.conclusionDate = null;
    this.maxDate = null;
    this.paidDate = null;
  }

  static fromJson(json: any): Invoice {
    const invoice = new Invoice();

    invoice.id = json.id;
    invoice.invoiceNumber = json.invoiceNumber;
    invoice.text = json.text;


    if (json.conclusionDate) invoice.conclusionDate = new Date(json.conclusionDate);
    if (json.maxDate) invoice.maxDate = new Date(json.maxDate);
    if (json.paidDate) invoice.paidDate = new Date(json.paidDate);


    return invoice;
  }

  static fromJsonList(jsonList: any[]): Invoice[] {
    return jsonList.map(json => Invoice.fromJson(json));
  }
}

File: 01-backend/src/main/java/ee/insa/crmapp/model/Invoice.java
package ee.insa.crmapp.model;


import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.time.LocalDate;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Entity
@Table(name = "invoice")
public class Invoice {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Long id;
    private String invoiceNumber;
    @Column(length = 1000)
    private String text;
    private LocalDate conclusionDate;
    private LocalDate maxDate;
    private LocalDate paidDate;
}